### PR TITLE
feat: GitHub request throttling

### PR DESCRIPTION
**Pull request by nwalters512**
This PR adds [`@octokit/plugin-throttling`](https://github.com/octokit/plugin-throttling.js), which will complement [`@octokit/plugin-retry`](https://github.com/octokit/plugin-retry.js) that we already use.

### CODE REVIEW

File: src/services/github.ts [hunk 1]
```diff
@@ -29,12 +30,16 @@ export default class GithubService {
         set a token on the 'password' field in ~/.netrc for api.github.com`);
       }
 
-      this.octokit = new RetryableOctokit({
+      this.octokit = new RetryableThrottledOctokit({
         auth: token,
-        retry: {
-          // By default the doNotRetry setting includes 403, which means we face secondary rate limits
-          doNotRetry: [400, 401, 404, 422],
-        }
```
Comment (nwalters512) [inline src/services/github.ts:37]: I dropped this since the new plugin should now handle `403` errors for us.

Comment (aorinevo) [reply src/services/github.ts:37]: Nice!

**Review by aorinevo — APPROVED** [review]

**Review by benjroy — APPROVED** [review]
Looks good to me

**Pull request merged**